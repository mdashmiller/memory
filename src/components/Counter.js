import React from 'react'
import styled from 'styled-components'

const CounterLabel = styled.h4`
	color: #bbc4ef;
	margin: 8px 0 0;
`

const StyledCounter = styled.div`
	width: 54px;
    height: 36px;
    margin: 4px auto 0;
    font-size: 32px;
    color: #bbc4ef;
    border: 2px solid #bbc4ef;
    border-radius: 4px;
`

const Counter = ({ moves }) => 
	<div>
		<CounterLabel id="label">MOVES</CounterLabel>
		<StyledCounter id="counter">
			{moves}
		</StyledCounter>
	</div>

export default Counter
