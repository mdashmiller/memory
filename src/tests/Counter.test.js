import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Counter from '../components/Counter'

describe('Counter', () => {

	const moves = '00'

	it('renders without crashing', () => {
		const div = document.createElement('div')
		ReactDOM.render(<Counter>{moves}</Counter>, div)
		ReactDOM.unmountComponentAtNode(div)
	})

	test('has a valid snapshot', () => {
		const component = renderer.create(
			<Counter>{moves}</Counter>
		)
		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

})
