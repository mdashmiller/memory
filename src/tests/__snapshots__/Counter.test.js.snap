// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter has a valid snapshot 1`] = `
<div>
  <h4
    className="sc-bdVaJa dDXGtt"
    id="label"
  >
    MOVES
  </h4>
  <div
    className="sc-bwzfXH bBXLIZ"
    id="counter"
  >
    00
  </div>
</div>
`;
